R/merge.rspec.R

```diff
@@ -1,51 +1,82 @@
-#' Merge two rspec objects
+#' Merge rspec objects
 #'
-#' Merges two `rspec` or `data.frame` objects into a single `rspec` object.
-#'
-#' @param x,y (required) `rspec` objects to merge.
-#' @param ... additional class arguments.
-#' @return an object of class `rspec` for use with `pavo` functions.
-#' Will use `by = "wl"` if unspecified, or automatically append `wl` to the
-#' `by` argument if one is specified.
+#' Merges two or more `rspec` objects into a single `rspec` object.
+#' 
+#' @param x (required) an `rspec` object
+#' @param ... (required) further `rspec` objects to merge by their wavelength ('wl') values, 
+#' and any additional parameters.
+#' @return an object of class `rspec` comprising multiple spectra with a single
+#' shared wavelength (`wl`) column, for use with `pavo` functions.
 #'
 #' @export
 #'
 #' @examples
 #'
 #' # Load angle-resolved reflectance data for a green-winged teal, and
-#' # split it in two
+#' # split it into three objects, each with a 'wl' column
 #' data(teal)
-#' teal1 <- teal[, c(1, 3:5)]
-#' teal2 <- teal[, c(1, 2, 6:12)]
+#' teal1 <- teal[, c(1, 2:5)]
+#' teal2 <- teal[, c(1, 6:8)]
+#' teal3 <- teal[, c(1, 9:13)]
 #'
-#' # Merge the two split datasets back into one, with a shared 'wl' column
-#' teal.mer <- merge(teal1, teal2, by = "wl")
+#' # Merge the three split datasets back into one, with a shared 'wl' column
+#' teal.mer <- merge(teal1, teal2, teal3)
 #'
 #' # Examine the results, and compare the original to the (identical)
 #' # reconstructed version
 #' plot(teal.mer)
 #' plot(teal)
+#' identical(teal.mer, teal)
 #'
+#' @author Thomas White \email{thomas.white026@@gmail.com}
 #' @author Chad Eliason \email{cme16@@zips.uakron.edu}
 #'
 #' @seealso [as.rspec()], [aggspec()]
 
-merge.rspec <- function(x, y, ...) {
-
-  if (!all(is.rspec(x), is.rspec(y))) {
-    stop("One or more invalid rspec objects")
-  }
-
-  if (!all("wl" %in% names(x), "wl" %in% names(y))) {
-    stop("Cannot find valid 'wl' column in one or both input objects")
+merge.rspec <- function(x, ...) {
```

Comment:
Even if we move forward with this, it's still necessary to include the `y` argument to match arguments in the generic.